include/swift/Parse/Token.h

```diff
@@ -266,17 +269,23 @@ class Token {
 
   /// \brief Set the token to the specified kind and source range.
   void setToken(tok K, StringRef T, unsigned CommentLength = 0,
-                bool MultilineString = false) {
+                bool MultilineString = false, unsigned DelimiterLength = 0) {
     Kind = K;
     Text = T;
     this->CommentLength = CommentLength;
     EscapedIdentifier = false;
     this->MultilineString = MultilineString;
+    StringDelimiterLength = DelimiterLength;
+    assert(StringDelimiterLength == DelimiterLength && "delimiter too long");
```

Comment:
This `assert` never hits.

Reply:
I does, honest. StringDelimiterLength is only 8 bits